shell/browser/browser_win.cc

```diff
@@ -584,8 +589,120 @@ v8::Local<v8::Promise> Browser::GetApplicationInfoForProtocol(
   return handle;
 }
 
-bool Browser::SetBadgeCount(int count) {
-  return false;
+int Browser::GetBadgeCount(base::Optional<bool> shouldGetBadge) {
+  if (!shouldGetBadge.has_value() || !shouldGetBadge.value()) {
+    return badge_count_;
+  } else {
+    return 0;
+  }
+}
+
+bool Browser::SetBadgeCount(base::Optional<int> count,
+                            base::Optional<bool> shouldSetBadge) {
+  if (!shouldSetBadge.has_value() || !shouldSetBadge.value()) {
+    // app.setBadgeCount(count) is not support on Windows, but
```

Comment:
I think it is fine to make `app.setBadgeCount` silently work on Windows, as long as it is an undocumented thing we are free to do whatever we want. So there is no need to add a special `shouldSetBadge` for Windows.